app/models/topic_list.rb

```diff
@@ -138,7 +138,17 @@ def load_topics
       { category: :parent_category },
     ]
 
-    topic_preloader_associations.concat(DiscoursePluginRegistry.topic_preloader_associations.to_a)
+    DiscoursePluginRegistry.topic_preloader_associations.each do |assoc|
+      if assoc.is_a?(Symbol)
+        topic_preloader_associations << assoc
+      elsif assoc.is_a?(Hash)
+        if assoc.key?(:association) && assoc.key?(:condition)
+          topic_preloader_associations << assoc[:association] if assoc[:condition].call
```

Comment:
Another thing to consider is that runtime conditions usually want to depend on runtime variables in which the `load_topics` is executed in. Should those be made available to the `condition` block as well? For example, the `@current_user`.